src/libraries/System.Private.Xml/src/System/Xml/Serialization/Xmlcustomformatter.cs

```diff
@@ -437,8 +437,7 @@ internal static char ToChar(string value)
         internal static byte[]? ToByteArrayHex(string? value)
         {
             if (value == null) return null;
-            value = value.Trim();
-            return XmlConvert.FromBinHexString(value);
+            return XmlConvert.FromBinHexString(value.AsSpan().Trim(XmlConvert.WhitespaceChars), true);
```

Comment:
this doesn't look equivalent - is WhitespaceChars same as default for Trim? I think Trim by default trims all Unicode whitespaces while WhitespaceChars doesn't seem like a complete list.

It could be that old code is wrong but I'd prefer we've fixed this separately if we can prove one way or another

Reply:
@krwq This is current implementation of XmlConverter.TrimString()
``` cs
// XML whitespace characters, <spec>http://www.w3.org/TR/REC-xml#NT-S</spec>
        internal static readonly char[] WhitespaceChars = new char[] { ' ', '\t', '\n', '\r' };

        // Trim a string using XML whitespace characters
        internal static string TrimString(string value)
        {
            return value.Trim(WhitespaceChars);
        }
```
There were 2 methods in XmlConvert which were converting string to byte hex and I replaced them with one which accepts Span

Before:
``` cs
internal static byte[] FromBinHexString(string s)
        {
            return FromBinHexString(s, true);
        }

        internal static byte[] FromBinHexString(string s, bool allowOddCount)
        {
            ArgumentNullException.ThrowIfNull(s);

            return BinHexDecoder.Decode(s.AsSpan(), allowOddCount);
        }
```
After
``` cs
internal static byte[] FromBinHexString(ReadOnlySpan<char> s, bool allowOddCount)
        {
            return BinHexDecoder.Decode(s, allowOddCount);
        }
```

Reply:
@krwq You are right - new implementation might differ. I will convert it back to string.AsSpan(),Trim().
If you like later can fix it to accept XmlConvert.WhitespaceChars

Reply:
yes, please file a separate issue to validate if this is correct behavior since this doesn't seem consistent with the rest of XML. We can think about this separately. Even if it makes sense to make this change it would be considered a breaking change and we should think about the trade-offs

Reply:
https://github.com/dotnet/runtime/issues/75523